provider/pkg/provider/provider.go

```diff
@@ -269,7 +269,9 @@ func (p *cfnProvider) Configure(ctx context.Context, req *pulumirpc.ConfigureReq
 		loadOptions = append(loadOptions, config.WithRegion(region))
 		p.region = region
 	} else {
-		return nil, errors.New("missing required property 'region'")
+		return nil, errors.New("missing required configuration key \"aws-native:region\":" +
+			"The region where AWS operations will take place. Examples are eu-east-1, eu-west-2, etc.\n" +
+			"\tSet a value using the command `pulumi config set aws-native:region <value>`")
```

Comment:
It might be better to link to https://www.pulumi.com/registry/packages/aws-native/installation-configuration/#setup for this. Setting the stack config is only one of several supported options for setting the region.

Reply:
It should also be able to pick up config from env vars without requiring the stack config, but we could address that in a follow-up PR.